docs/app-dev-context/activity-parameters.md

```diff
@@ -1,4 +1,19 @@
-All Activity parameters must be serializable.
+---
+id: activity-parameters
+title: Activity Parameters - Application development context
+sidebar_label: Activity Parameters
+description: When it comes to your application data—that is, data that is serialized and encoded into a Payload—we recommend that you use a single object as an argument that wraps the application data passed to Activities.
+tags:
+  - guide-context
+---
 
-There is no explicit limit to the amount of parameter data that can be passed to an Activity, but keep in mind that all parameters and return values are recorded in a [Workflow Execution Event History](/concepts/what-is-an-event-history).
-A large Workflow Execution Event History can adversely impact the performance of your Workflow Executions, because the entire Event History is transferred to Worker Processes with every [Workflow Task](/concepts/what-is-a-workflow-task).
+There is no explicit limit to the total number of parameters that an [Activity Definition](/concepts/what-is-an-activity-definition) may support.
```

Comment:
@bergundy @lorensr I would love your feedback to help make this correct and complete -

Reply:
Looks good, would just add for now max Payload/single-argument size 2mb, max grpc message (which includes all the arguments) 4mb